lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -47,9 +47,47 @@ struct WhileScheduleable {
   calyx::GroupOp initGroup;
 };
 
-// A variant of types representing scheduleable operations.
+/// A variant of types representing scheduleable operations.
 using Scheduleable = std::variant<calyx::GroupOp, WhileScheduleable>;
 
+/// A structure representing a set of ports which act as a memory interface.
+struct CalyxMemoryPorts {
+  Value readData;
+  Value done;
+  Value writeData;
+  SmallVector<Value> addrPorts;
+  Value writeEn;
+};
+
+/// The various lowering passes are agnostic wrt. whether working with a
+/// calyx::MemoryOp (internally allocated memory) or external memory (through
+/// CalyxMemoryPort). This is achieved through the following
+/// CalyxMemoryInterface for accessing either a calyx::MemoryOp or a
+/// CalyxMemoryPorts struct.
+struct CalyxMemoryInterface {
```

Comment:
One future thought is pulling this out into some kind of utility file, as I can see this being useful for any frontend that uses memories.